part-2.py

```diff
@@ -4,13 +4,41 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query, n = 0):
+    if not len(array):
+        return False
+    if n >= len(array):
+        return False
+    if array[n] == query:
+        return True
+        
+    return search(array, query, n+1)
 
 
 # is_palindrome
+def is_palindrome(text):
+    
+    
+    if(text == text[::- 1]):
+        return True
+    else:
+        return False
+
+    is_palindrome(text)
```

Comment:
so you are calling this recursively, but `text[::-1]` is doing all the work. It is calling a for loop and re-concatenating the string. 

What we want to do is have our function check the starting index and the last index to see if they are equal. If they are, we call `is_palindrome` again and decrement the last index, increment the starting index and then check again.